hbase_storage/src/main/java/com/splicemachine/si/impl/region/TxnTimeTraveler.java

```diff
@@ -0,0 +1,153 @@
+            Pair<Long, Long> middleTx;
+            middleTx = txnFinder.find(bucket, middle, reversed);
+            reversed = false;
+            if(LOG.isDebugEnabled())
+                SpliceLogUtils.debug(LOG,"middle of range: [%s (TS: %s) id %s, %s (TS: %s)] id %s is %s (TS: %s) id %s",
+                        TxnUtils.rowKeytoHbaseEscaped(left), new Timestamp(leftTS).toString(), txOf(left),
+                        TxnUtils.rowKeytoHbaseEscaped(right), new Timestamp(rightTS).toString(), txOf(right),
+                        TxnUtils.rowKeytoHbaseEscaped(middle), new Timestamp(middleTx.getSecond()).toString(), middleTx.getFirst().toString());
+            long middleTS = middleTx.getSecond();
+            if(middleTS == needleTS) {
+                return middleTx;
+            } else if(needleTS < middleTS) {
+                if(middleTS == rightTS)
+                {
+                    reversed = true;
+                    changed = true;
+                }
+                else if(middleTS < rightTS)
+                {
+                    right = toRowKeyWithBucket(middleTx.getFirst(), bucket);
+                    rightTS = middleTS;
+                    changed = true;
+                }
+            } else if(leftTS < middleTS) {
+                left = toRowKeyWithBucket(middleTx.getFirst(), bucket);
+                leftTS = middleTS;
+                changed = true;
+            }
+        }
+        return new Pair<>(txOf(left), leftTS);
+    }
+
+    private long txOf(byte[] rowKey) {
+        return TxnUtils.txnIdFromRowKey(rowKey, 0, rowKey.length);
+    }
+
+    private byte[] middle(byte[] left, byte[] right) {
+        assert(left.length == 9); assert(right.length == 9);
+        byte lBucket = left[0], rBucket = right[0];
+        assert(lBucket == rBucket);
+        long lTxnId = txOf(left), rTxnId = txOf(right), middleTxnId = (lTxnId + rTxnId) / 2;
+        return toRowKeyWithBucket(middleTxnId, lBucket);
+    }
+
+    private Pair<Long, Long> first(byte bucket) throws IOException {
+        return txnFinder.find(bucket, null,false);
+    }
+
+    private Pair<Long, Long> last(byte bucket) throws IOException {
+        return txnFinder.find(bucket, null,true);
+    }
+
+    private byte[] toRowKeyWithBucket(long txId, byte bucket) {
+        byte[] result = TxnUtils.getRowKey(txId);
+        assert result.length == 9 : "unexpected rowKey format";
+        result[0] = bucket;
+        return result;
+    }
+
+    // When we use a past transaction P whose timestamp is T, we can read data which is committed before T
+    // i.e. all data with time < T will be visible, therefore, to be able to read the data of the past
+    // transaction itself, we slide to the next tx.
+    private long slideToNextTx(long tx) throws IOException {
+        if(tx == -1) { // special case for timestamp that happened before the first transaction.
+            return -1;
+        }
+        return tx + SIConstants.TRASANCTION_INCREMENT;
```

Comment:
While  misspellings generally aren't noteworthy and this isn't a blocker for merging today, the literal here is spelled wrong. Could make searching for it harder in the future, so can we fix this in a future release?